raiden/network/rpc/client.py

```diff
@@ -172,83 +192,61 @@ def __init__(
             port: int,
             privkey: bytes,
             gasprice: int = None,
-            nonce_update_interval: float = 5.0,
             nonce_offset: int = 0,
             web3: Web3 = None,
     ):
-
         if privkey is None or len(privkey) != 32:
             raise ValueError('Invalid private key')
 
         endpoint = 'http://{}:{}'.format(host, port)
+        web3: Web3 = web3 or Web3(HTTPProvider(endpoint))
 
+        monkey_patch_web3(web3, self)
+
+        try:
+            version = web3.version.node
+        except ConnectTimeout:
+            raise EthNodeCommunicationError('couldnt reach the ethereum node')
+
+        supported, eth_node = is_supported_client(version)
+
+        if not supported:
+            print('You need a Byzantium enabled ethereum node. Parity >= 1.7.6 or Geth >= 1.7.2')
+            sys.exit(1)
```

Comment:
As discussed yesterday perhaps raise an exception here and catch it outside and do the sys.exit() and printing an error message there?

Reply:
I will merge it and we can consider this in a later PR.